subprojects/docs/src/docs/userguide/jvm/java_library_plugin.adoc

```diff
@@ -241,6 +241,146 @@ The role of each configuration is described in the following tables:
+
+A third case are traditional libraries that provide no module information at all -- for example `commons-cli:commons-cli:1.4`.
+Gradle puts such libraries on the classpath instead of the module path.
+The classpath is then treated as one module (the so called _unnamed_ module) by Java.
+
+.Dependencies to modules and libraries declared in build file
+====
+include::sample[dir="snippets/java-library/module/groovy",files="build.gradle[tags=dependencies]"]
+include::sample[dir="snippets/java-library/module/kotlin",files="build.gradle.kts[tags=dependencies]"]
+====
+
+[source,java,indent=0]
+.Module dependencies declared in module-info.java file
+----
+include::{samplesPath}/java-library/module/kotlin/src/main/java/module-info.java[]
+----
+
+While a real module cannot directly depend on the unnamed module (only by adding command line flags), automatic modules can also see the unnamed module.
+Thus, if you cannot avoid to rely on a library without module information, you can wrap that library in an automatic module as part of your project.
+How you do that is described in the next section.
+
+Another way to deal with non-modules is to enrich existing Jars with module descriptors yourself using <<artifact_transforms.adoc#,artifact transforms>>.
+link:../samples/sample_java_modules_with_transform.html[This sample] contains a small _buildSrc_ plugin registering such a transform which you may use and adjust to your needs.
+This can be interesting if you want to build a fully <<application_plugin.adoc#sec:application_modular,modular application>> and want the java runtime to treat everything as a real module, e.g. for performance considerations.
+
+=== Building an automatic module
+
+If you can, you should always write complete `module-info.java` descriptors for your modules.
+Still, there are a few cases where you might consider to (initally) only provide a _module name_ for an automatic module:
+
+- You are working on a library that is **not** a module but you want to make it usable as such in the next release.
+  Adding an `Automatic-Module-Name` is a good first step (most popular OSS libraries on Maven central have done it by now).
+- As discussed in the previous section, an automatic module can be used as an adapter between your real modules and a traditional library on the classpath.
+
+To turn a normal Java project into an _automatic module_, just add the manifest entry with the module name:
+
+.Declare an automatic module name as Jar manifest attribute
+====
+include::sample[dir="snippets/java-library/module/groovy",files="build.gradle[tags=automaticModuleName]"]
+include::sample[dir="snippets/java-library/module/kotlin",files="build.gradle.kts[tags=automaticModuleName]"]
+====
+
+NOTE: You can define an automatic module as part of a multi-project that otherwise defines real modules (e.g. as an adapter to another library).
```

Comment:
Let's be consistent with asciidoc features and use:

>[NOTE]
>\====
>text
>\====

Reply:
So that is what we _should_ use? I'll change it.  The `NOTE:` notation is already all over the place as well. :) If we want to get rid of it, we should adjust all the paces at some point.

Reply:
I'll adjust all occurrences in the JVM part of the docs for now.